src/ray/core_worker/experimental_mutable_object_manager.h

```diff
@@ -36,72 +36,71 @@ class MutableObjectManager {
  public:
   struct Channel {
     Channel(std::unique_ptr<plasma::MutableObject> mutable_object_ptr)
-        : mutable_object(std::move(mutable_object_ptr)) {}
-
-    std::unique_ptr<plasma::MutableObject> mutable_object;
-  } ABSL_CACHELINE_ALIGNED;
-
-  struct WriterChannel : public Channel {
-    WriterChannel(std::unique_ptr<plasma::MutableObject> mutable_object_ptr)
-        : Channel(std::move(mutable_object_ptr)) {}
+        : lock(std::make_unique<absl::Mutex>()),
+          mutable_object(std::move(mutable_object_ptr)) {}
 
     // WriteAcquire() sets this to true. WriteRelease() sets this to false.
     bool written = false;
-  };
+    bool read = false;
```

Comment:
Can you add a comment here to explain what it's used for? Also, maybe it should be `reading`?